### PR TITLE
Use faster JS selection methods

### PR DESCRIPTION
**Pull request by rhymes**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description

While reviewing another PR I noticed how we tend to use interchangeably various selectors like `getElementsByClassName` or `querySelector*` but the first, older bunch, is almost always faster than the latter. The `querySelector*` group should be favored only when actual CSS selectors (other than ID, class name and tag name) are needed. 

I also added a "tips" section to the frontend documentation

## Related Tickets & Documents

https://github.com/forem/forem/issues/6380

## QA Instructions, Screenshots, Recordings

Click around

## Added tests?

- [ ] Yes
- [x] No, and this is why: because tests need to pass regardless of the change, they are our regression
- [ ] I need help with writing tests

## Added to documentation?

- [x] [Developer Docs](https://docs.forem.com) and/or [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [ ] README
- [ ] No documentation needed

### CODE REVIEW

**Review by rhymes — COMMENTED** [review]
I know this PR touches many files but it's just mainly the following things:

- replace `querySelectorAll('.className')` with `getElementsByClassName('className')`
- replace `querySelector('.className')` with `getElementsByClassName('className')[0]`
- replace `querySelector*('tagName')` with `getElementsByTagName('tagName')`
- replace `querySelector('#id')` with `getElementById('id')`

File: app/assets/javascripts/initializers/initializeEllipsisMenu.js [hunk 1]
```diff
@@ -47,7 +47,9 @@ function onXhrSuccess(form, article, values) {
     toggleNotifications(submit, submitValue);
   }
 
-  article.querySelector('.js-ellipsis-menu').classList.add('hidden');
+  article
+    .getElementsByClassName('.js-ellipsis-menu')[0]
```
Comment (Ridhwana) [inline app/assets/javascripts/initializers/initializeEllipsisMenu.js:51]: @rhymes should this be:

```suggestion
    .getElementsByClassName('js-ellipsis-menu')[0]
```

File: app/assets/javascripts/utilities/slideSidebar.js [hunk 2]
```diff
@@ -18,7 +16,7 @@ function slideSidebar(side, direction) {
     document.body.classList.remove('modal-open');
     document
       .getElementById('sidebar-wrapper-' + side)
-      .querySelector('.side-bar').scrollTop = 0;
+      .getElementsByClassName('.side-bar')[0].scrollTop = 0;
```
Comment (Ridhwana) [inline app/assets/javascripts/utilities/slideSidebar.js:19]: Should it be?
```suggestion
      .getElementsByClassName('side-bar')[0].scrollTop = 0;
```

**Review by Ridhwana — COMMENTED** [review]
@rhymes I'm halfway though the review. I stopped by actionsPanel.js, but thought I'd put out these comments already, will review the rest of the PR in a bit.

**Comment by rhymes** [comment]
Thanks @Ridhwana, really appreciate it! I ran `rg getElementsByClassName | rg "'\."` and I can't find any others

**Review by Ridhwana — APPROVED** [review]
I went through the files and this PR looks good @rhymes. Nice touch with the documentation as well 🎉

**Review by msarit — APPROVED** [review]
TIL:  `getElementsBy*` is almost always faster than `querySelector*` 🙏

**Pull request merged**